File: 2ºAÑO/1ºCuatrimestre/PDOO/examen_p1_p2/examen/lib/TestP1.rb
# encoding:utf-8

require_relative "dado.rb"
require_relative "tablero.rb"
require_relative "sorpresa.rb"
require_relative "estados_juego.rb"
require_relative "mazo_sorpresas.rb"
require_relative "diario.rb"

module Civitas
  class TestP1
    def self.main
      
      @@TAMANIO=20
      
      puts " \n\n*****TAREA 1*****"
      
      uno = 0
      dos = 0
      tres = 0
      cuatro = 0
      
      for i in(1..100)
        
        empieza = Dado.instance.quien_empieza(4)+1
        
        if empieza == 1
          uno +=1
        end
        
        if empieza == 2
          dos +=1
        end
        
        if empieza == 3
          tres +=1
        end
        
        if empieza == 4
          cuatro +=1
        end
      end
      
      puts "El jugador 1 ha salido #{uno} veces"
      puts "El jugador 2 ha salido #{dos} veces"
      puts "El jugador 3 ha salido #{tres} veces"
      puts "El jugador 4 ha salido #{cuatro} veces"
      
      
      puts "\n\n*****TAREA 2*****"
      
      Dado.instance.debug=(true)
      puts Dado.instance.tirar
      
      Dado.instance.debug=(false)
      puts Dado.instance.tirar
      
      puts"\n\n*****TAREA 3*****"
      
      puts Dado.instance.tirar
      puts Dado.instance.get_ultimo_resultado
      puts Dado.instance.salgo_de_la_carcel
      
      puts "\n\n*****TAREA 4*****"
      
      puts "Casilla de tipo " + Tipo_casilla::CALLE.to_s
      puts "Sorpresa de tipo " + Tipo_sorpresa::SALIRCARCEL.to_s
      puts "Estado de juego " + Estados_juego::INICIO_TURNO.to_s
      
      puts "\n\n**TAREA 5**"
      prueba = ""
      @@mazo = Mazo_sorpresas.new
      @@s1 = Sorpresa.new_generico(Tipo_sorpresa::IRCARCEL,25,"hola2")
      @@s2 = Sorpresa.new_generico(Tipo_sorpresa::IRCARCEL,25,"hola2")
      @@mazo.al_mazo(@@s1)
      @@mazo.al_mazo(@@s2)
      @@mazo.siguiente
      @@mazo.inhabilitar_carta_especial(@@s2)
      @@mazo.habilitar_carta_especial(@@s2)
      puts Diario.instance.leer_evento
      puts Diario.instance.ocurre_evento("prueba")
      puts Diario.instance.leer_evento
      puts Diario.instance.eventos_pendientes()
            
      puts "\n\n **TAREA 7**"
      carcel = 0
      
      puts "Introduce el nuero de casilla de carcel: "
      carcel = gets.to_i 
      
      calle1 = "CALLE 1"
      calle2 = "CALLE 2"
      calle3 = "CALLE 3"
      calle4 = "CALLE 4"
      calle5 = "CALLE 5"
      calle6 = "CALLE 6"
      calle7 = "CALLE 7"
      calle8 = "CALLE 8"
      calle9 = "CALLE 9"
      
      @@tablero = Tablero.new(carcel)
      c1 = Casilla.new_casilla_descanso(calle1)
      c2 = Casilla.new_casilla_descanso(calle2)
      c3 = Casilla.new_casilla_descanso(calle3)
      c4 = Casilla.new_casilla_descanso(calle4)
      c5 = Casilla.new_casilla_descanso(calle5)
      c6 = Casilla.new_casilla_descanso(calle6)
      c7 = Casilla.new_casilla_descanso(calle7)
      c8 = Casilla.new_casilla_descanso(calle8)
      c9 = Casilla.new_casilla_descanso(calle9)
        
      @@tablero.añade_casilla(c1)
      @@tablero.añade_casilla(c2)
      @@tablero.añade_casilla(c3)
      @@tablero.añade_casilla(c4)
      @@tablero.añade_casilla(c5)
      @@tablero.añade_casilla(c6)
      @@tablero.añade_casilla(c7)
      @@tablero.añade_casilla(c8)
      @@tablero.añade_casilla(c9)
      
      if carcel > @@TAMANIO
        puts "ERROR, la posicion de carcel es mayor que el tamaño del tablero \n"
        @@tablero.añade_casilla(Casilla.new("Carcel"))
      else
        puts "Tablero creado correctamente"
      end
      
      @@tablero.añade_juez
      
      origen = 7
      
      puts "El jugador se encuentra en la casilla 7"
      valor_dado = Dado.instance.tirar
      puts "Tiramos el dado y sale #{valor_dado}"
      puts "Por lo tanto el jugador avanza hasta la casilla " + @@tablero.nueva_posicion(origen, valor_dado).to_s
      
      #hay 9 casillas + juez y mas carcel
      
    end
  end
  TestP1.main
end

File: 2ºAÑO/1ºCuatrimestre/PDOO/Civitas/Civitas/src/civitas/TestP1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.Scanner;

/**
 *
 * @author Jose
 */
public class TestP1 {

    /**
     * @param args the command line arguments
     */
    public static void main(String[] args) {
        //TAREA 1
        System.out.println("\n\n *****TAREA 1*****");
        
        int tamanio = 20;
        
        int uno = 0;
        int dos = 0;
        int tres= 0;
        int cuatro = 0;
        
        for(int i = 1; i <= 100; i++) {
            int empieza = Dado.getInstance().quienEmpieza(4)+1;
            
            if(empieza == 1) {
                uno++;
            }
            
            else if(empieza == 2) {
                dos++;
            }
            
            else if(empieza == 3) {
                tres++;
            }
            
            else if(empieza == 4) {
                cuatro++;
            }
        }
        
        System.out.println("El jugador 1 ha salido " + uno + " veces");
        System.out.println("El jugador 2 ha salido " + dos + " veces");
        System.out.println("El jugador 3 ha salido " + tres + " veces");
        System.out.println("El jugador 4 ha salido " + cuatro + " veces");
        System.out.println("LA SUMA DE CADA VALOR DA: " + (uno + dos + tres + cuatro));
        
        //TAREA 2
        System.out.println("\n\n *****TAREA 2*****");
        Boolean deb = true;
        Dado.getInstance().setDebug(deb);
        System.out.println("Modo debug activado");
        int valor = Dado.getInstance().tirar();
        System.out.println("Valor dado: " + valor);
        
        deb = false;
        Dado.getInstance().setDebug(deb);
        System.out.println("Modo debug desactivado");
        valor = Dado.getInstance().tirar();
        System.out.println("Valor dado: " + valor);
        
        //TAREA 3
        System.out.println("\n\n *****TAREA 3*****");
        System.out.println("Tiramos el dado y sale: " + Dado.getInstance().tirar());
        System.out.println("El último resultado del dado es " + Dado.getInstance().getUltimoResultado());
        if(Dado.getInstance().salgoDeLaCarcel()) {
            System.out.println("Como ha salido mayor o igual que 5, salimos de la cárcel");
        }
        
        else {
            System.out.println("Ha salido menor que 5, por lo tanto no salimos de la cárcel");
        }
        
        //TAREA 4
        System.out.println("\n\n *****TAREA 4*****");
        System.out.println("Casilla de tipo " + TipoCasilla.CALLE);
        System.out.println("Sorpresa de tipo " + TipoSorpresa.SALIRCARCEL);
        System.out.println("Estado de Juego: " + EstadosJuego.INICIO_TURNO);
        
        //TAREA 5
        System.out.println("\n\n *****TAREA 5*****");
        String prueba = "";
        MazoSorpresas mazo = new MazoSorpresas();
        Sorpresa s1 = new Sorpresa(TipoSorpresa.SALIRCARCEL, mazo);
        Sorpresa s2 = new Sorpresa(TipoSorpresa.PORJUGADOR, 12, "Hoy es el dia de la felicidad, todos los jugadores te dan 12");
        mazo.alMazo(s1);
        mazo.alMazo(s2);
        mazo.siguiente();
        mazo.inhabilitarCartaEspecial(s2);
        mazo.habilitarCartaEspecial(s2);
        //Esto lo hemos hecho con breakproints y debugere
        System.out.println(Diario.getInstance().leerEvento());
        Diario.getInstance().ocurreEvento("prueba");
        System.out.println(Diario.getInstance().leerEvento());
        System.out.println(Diario.getInstance().eventosPendientes());
        
        //TAREA 7
        System.out.println("\n\n *****TAREA 7*****");
        Scanner capt = new Scanner(System.in);
        
        System.out.println("\n\n Dime el numero de la carcel");
        int carcel = capt.nextInt();
        
        
        String calle1 = "CALLE 1";
        String calle2 = "CALLE 2";
        String calle3 = "CALLE 3";
        String calle4 = "CALLE 4";
        String calle5 = "CALLE 5";
        String calle6 = "CALLE 6";
        String calle7 = "CALLE 7";
        String calle8 = "CALLE 8";
        String calle9 = "CALLE 9";
        
        Tablero tablero = new Tablero(carcel);
        Casilla c1 = new Casilla(calle1);
        Casilla c2 = new Casilla(calle2);
        Casilla c3 = new Casilla(calle3);
        Casilla c4 = new Casilla(calle4);
        Casilla c5 = new Casilla(calle5);
        Casilla c6 = new Casilla(calle6);
        Casilla c7 = new Casilla(calle7);
        Casilla c8 = new Casilla(calle8);
        Casilla c9 = new Casilla(calle9);
        
        tablero.añadeCasilla(c1);
        tablero.añadeCasilla(c2);
        tablero.añadeCasilla(c3);
        tablero.añadeCasilla(c4);
        tablero.añadeCasilla(c5);
        tablero.añadeCasilla(c6);
        tablero.añadeCasilla(c7);
        tablero.añadeCasilla(c8);
        tablero.añadeCasilla(c9);
        
        if(carcel > tamanio) {
            System.out.println("ERROR, la posicion de carcel es mayor que el tamaño del tablero \n");
            tablero.añadeCasilla(new Casilla("Carcel"));
        }
        
        else {
            System.out.println("Tablero creado correctamente \n");
        }
        
        tablero.añadeJuez();
        
        int origen = 7;
        System.out.println("El jugador se encuentra en la casilla 7");
        int valor_dado = Dado.getInstance().tirar();
        System.out.println("Tiramos el dado y sale: " + valor_dado);
        System.out.println("Por lo tanto el jugador avanza hasta la casilla " + tablero.nuevaPosicion(origen, valor_dado));
        //hay 9 casillas + juez y mas carcel
        //hacer los metodos to_s
        
    }  
}
